telegram/bot.py

```diff
@@ -3690,10 +3696,49 @@ def unpin_chat_message(
         """
         data: JSONDict = {'chat_id': chat_id}
 
+        if message_id is not None:
+            data['message_id'] = message_id
+
         result = self._post('unpinChatMessage', data, timeout=timeout, api_kwargs=api_kwargs)
 
         return result  # type: ignore[return-value]
 
+    @log
+    def unpin_all_chat_messages(
+        self,
+        chat_id: Union[str, int],
+        timeout: float = None,
+        api_kwargs: JSONDict = None,
+    ) -> bool:
+        """
+        Use this method to remove a message from the list of pinned messages in a chat. If the
+        chat is not a private chat, the bot must be an administrator in the chat for this
+        to work and must have the ``can_pin_messages`` admin right in a supergroup or
+        ``can_edit_messages`` admin right in a channel.
+
+        Args:
+            chat_id (:obj:`int` | :obj:`str`): Unique identifier for the target chat or username
+                of the target channel (in the format @channelusername).
+            timeout (:obj:`int` | :obj:`float`, optional): If this value is specified, use it as
+                the read timeout from the server (instead of the one specified during creation of
+                the connection pool).
+            api_kwargs (:obj:`dict`, optional): Arbitrary keyword arguments to be passed to the
+                Telegram API.
+
+        Returns:
+            :obj:`bool`: On success, :obj:`True` is returned.
+
+        Raises:
+            :class:`telegram.TelegramError`
+
+        """
+
+        data: JSONDict = {'chat_id': chat_id}
+
+        result = self._post('unpinAllChatMessages', data, timeout=timeout, api_kwargs=api_kwargs)
```

Comment:
you can just `return self._post(…)`

Reply:
That `result = ...` style is being used everywhere else too. Change those occurrences too?

Reply:
no need, it's not too important. just noticed here, so we can change it here :)

Reply:
the `# type: ignore[return-value]` thing doesn't work after doing that (hence failing tests), am I doing it wrong or should I revert?

Reply:
Nvm fixed it, that thing is weird